src/extension/content.ts

```diff
@@ -188,23 +215,45 @@ function getProjectInfo(pathName){
   const repoURLName = repoURLIndex.join('/');
   return { mergeRequestID: pathArray.at(midIndex+1), repoURLName};
 }
-const main = () => {
-  log("init");
-  const pathName = window.location.pathname;
-  const projectInfo = getProjectInfo(pathName);
-  const interval = setInterval(async () => {
+async function renderWidget(projectInfo){
+  let retryCounter = 1;
+  while(retryCounter<=2){
     try{
       let res = await getMergeRequestInfo(projectInfo.repoURLName,projectInfo.mergeRequestID);
-      clearInterval(interval);
       if(!res.isMerged){
-        initialise(projectInfo.repoURLName,projectInfo.mergeRequestID,res.sourceBranch,res.targetBranch);
+        initialise(projectInfo.repoURLName,projectInfo.mergeRequestID,res.sourceBranch,res.targetBranch,res.isRebaseInProgress);
+        return;
       }
     }catch(e){
-      console.log(e);
+        console.log(e);
     }
-  }, 1000);
+    await wait(2000);
+    retryCounter+=1;
+  }
+}
+const main = () => {
+  log("init");
+  const pathName = window.location.pathname;
+  const projectInfo = getProjectInfo(pathName);
+  
+  const targetNode = document.querySelector('.issuable-discussion');
+  const config = { childList: true, subtree: true };
+
+  const callback = function(mutationList, observer) {
```

Comment:
will this work if the widget is loaded after the mr-widget-section has already been added to the dom ?

Reply:
Yes, there is a issue. We have to check if section has already been loaded